### DISCUSSION TITLE
Static Generation and Cache-Control

### DISCUSSION THREAD

**Discussion by Fenntasy** [vercel/next.js#25134] [Help] [answered]
Hi,

I have a simple website that is right now totally static, there's no need for `getServerSideProps`, `getStaticProps` nor `getStaticPaths` right now. It's hosted on Vercel and I'm trying to understand how the cache is managed.

Right now, my pages have: `cache-control: public, max-age=0, must-revalidate` which is not ideal for static pages.

I've tried adding an empty `getStaticProps` function and even tried to add a `revalidate` param to see if that changes something and with an empty function, it returns a `cache-control: s-maxage=0` which is better but I don't understand how it's supposed to work.

The only solution I found to have a `max-age=3600` (just an example value) is to use a `getServerSideProps` function that only set the header. This seems to work locally but ~is ignored on Vercel~ (edit: it takes time to take effect).

I don't see how I can make it work without having this `must-revalidate` header that essentially disable all cache when combined with with `max-age=0` :/ 

Is there something I'm missing, shouldn't it be cached by default when there is no server props function?

**Comment by thanhtoan1196** [comment]
any updates?

**Comment by leerob** [comment, marked as answer]
It's now possible to customize `cache-control` as of v14.2.10. Please upgrade and give it a try!

https://github.com/vercel/next.js/pull/69802
